sycl/source/detail/platform_impl.cpp

```diff
@@ -49,6 +50,109 @@ platform_impl_host::get_devices(info::device_type dev_type) const {
   return res;
 }
 
+struct DevDescT {
+  const char *devName = nullptr;
+  int devNameSize = 0;
+
+  const char *minDriverVer = nullptr;
+  int minDriverVerSize = 0;
+};
+
+static std::vector<DevDescT> getWhiteListDesc() {
+  // TODO: Replace with const char *str =
+  // SYCLConfig<SYCL_DEVICE_WHITE_LIST>::get();
+  const char *str = getenv("SYCL_DEVICE_WHITE_LIST");
+  if (!str)
+    return {};
+
+  std::vector<DevDescT> decDescs;
+  const char devNameStr[] = "DeviceName";
+  const char driverVerStr[] = "DriverVersion";
+  decDescs.emplace_back();
+  while (*str != '\0') {
+    const char **dstPtr = nullptr;
+    int *Size = nullptr;
+
+    // -1 to avoid comparing null terminator
+    // TODO: Change devNameStr so it doesn't contain null terminator
+    if (0 == strncmp(devNameStr, str, sizeof(devNameStr) - 1)) {
+      dstPtr = &decDescs.back().devName;
+      Size = &decDescs.back().devNameSize;
+      str += sizeof(devNameStr);
```

Comment:
What about `:`? According to commit message, the format is `DeviceName:XXX,DriverVersion:X.Y.Z.W`, so, you need to skip `:` symbol or it will be included into a device name

Reply:
It's kinda skipped here as sizeof(devNameStr) counts null terminator.
Will change the code so it's done explicitly.

Reply:
Done.